### PR TITLE
feat: implement `appDir`

### PR DESCRIPTION
**Pull request by stipsan**
# [Preview Deploy link](https://vercel.com/new/clone?repository-url=https%3A%2F%2Fgithub.com%2Fsanity-io%2Fnextjs-blog-cms-sanity-v3%2Ftree%2Frsc&repository-name=blog-nextjs-sanity&project-name=blog-nextjs-sanity&demo-title=Blog%20with%20Built-in%20Content%20Editing&demo-description=A%20Sanity-powered%20blog%20with%20built-in%20content%20editing%20%26%20instant%20previews&demo-url=https%3A%2F%2Fnextjs-blog.sanity.build%2F%3Futm_source%3Dvercel%26utm_medium%3Dreferral&demo-image=https%3A%2F%2Fuser-images.githubusercontent.com%2F81981%2F197501516-c7c8092d-0305-4abe-afb7-1e896ef7b90a.png&integration-ids=oac_hb2LITYajhRQ0i4QznmKH7gx&external-id=nextjs;template=nextjs-blog-cms-sanity-v3)

~~Blocked by:~~
- ~~All Document/HTML requests are always missing cache, verify by looking at `x-vercel-cache` which is `MISS` but should be `HIT` like it is on the `main`.~~
- ~~When in production (`next build && next start`) and `SANITY_REVALIDATE_SECRET` isn't set, an error is thrown when trying to view a post:~~
  ```bash
  Error: invariant: invalid Cache-Control duration provided: 0 < 1
  ```
- ~~When running locally (`next dev`) the post pages throw this error:~~
  ```bash
  Error: Dynamic server usage: previewData
  ```

`previewData` blockers worked around, revisit in https://github.com/sanity-io/nextjs-blog-cms-sanity-v3/issues/95 once Next 13 is shipping an updated `previewData()` mechanic.

### CODE REVIEW

**Comment by kmelve** [comment]
#### 📝 Changed routes:




2 deleted routes:
- **/**
- **/posts/[slug]**

Commit ab9cb5f5ecf0a3b853da6120d34afcf378524a76 (https://nextjs-blog-cms-sanity-v3-rlnp5jo7t.sanity.build).

**Comment by arduano** [comment]
Hi, with the cache control error that you listed as blocking, I was debugging the same error yesterday in my own project and I traced it down to a single `usePathname` call when ISR is enabled and a non cached page is fetched. I double checked entirely in isolation in a new test project to reproduce it. Also the error went away when I updated nextjs and react to the latest versions too. Your cause may be different but hopefully this helps.

**Comment by stipsan** [comment]
@arduano in this case it's calling `previewData()` that causes the [bailout](https://github.com/vercel/next.js/blob/3a770fd1438475b7c08c71de42a8e37451dd4966/packages/next/client/components/headers.ts#L19).

If you remove all the `previewData()` calls then things work like they do on main. You'll get a 60min TTL on the home page and slug pages, where it regens in the background and returns a `STALE` header whenever the TTL expires and you make a request.
Here's a [deploy](https://no-preview-data.vercel.app/) for you to check the cache headers.

File: app/head.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
...41NjY5MyAxLjE4MTE2IDQuMzI3ODEgMS45NjEyNiAzLjMxNDc5IiBmaWxsPSJjdXJyZW50Q29sb3IiPjwvcGF0aD48cGF0aCBvcGFjaXR5PSIwLjciIGQ9Ik01Mi45ODI0IDEzLjY0MTVWMS4wMDcxNEg1Ny4wNjAyVjIxLjAwMjhINTIuOTgyNFYxMy42NDE1WiIgZmlsbD0iY3VycmVudENvbG9yIj48L3BhdGg%2BPHBhdGggb3BhY2l0eT0iMC43IiBkPSJNMTIuNzQ1OCAxNC4zNjg5QzE0LjMyOTQgMTUuMzY0MyAxNS4wMjM4IDE2Ljc1NjUgMTUuMDIzOCAxOC43NTQ0QzEzLjcxMyAyMC40MDQxIDExLjQxMDEgMjEuMzMgOC43MDMzMyAyMS4zM0M0LjE0NzE4IDIxLjMzIDAuOTU4NTc3IDE5LjEyNjggMC4yNSAxNS4yOTgySDQuNjI1NDdDNS4xODg3OCAxNy4wNTU5IDYuNjgwMzQgMTcuODcwMyA4LjY3MTQ0IDE3Ljg3MDNDMTEuMTAxOSAxNy44NzAzIDEyLjcxNzQgMTYuNTk2NCAxMi43NDkzIDE0LjM2MTkiIGZpbGw9ImN1cnJlbnRDb2xvciI%2BPC9wYXRoPjxwYXRoIG9wYWNpdHk9IjAuNyIgZD0iTTQuMjM1NjcgNy40NDI2N0MzLjUxMjUgNy4wMjA0NSAyLjkxOTIgNi40MTM3NSAyLjUxODczIDUuNjg2OTdDMi4xMTgyNyA0Ljk2MDE5IDEuOTI1NTggNC4xNDA0NSAxLjk2MTEzIDMuMzE0NzZDMy4yMjU5NCAxLjY3ODkxIDUuNDI2MDggMC42Nzk5OTMgOC4xMDgwNCAwLjY3OTk5M0MxMi43NDkyIDAuNjc5OTkzIDE1LjQzNDcgMy4wODg1MiAxNi4wOTcyIDYuNDc4NTZIMTEuODg4M0MxMS40MjQyIDUuMTQyMDMgMTAuMjYyMSA0LjEwMTM2IDguMTQzNDcgNC4xMDEzNkM1Ljg3OTU3IDQuMTAxMzYgNC4zMzQ4NyA1LjM5NjExIDQuMjQ2MjkgNy40NDI2NyIgZmlsbD0iY3VycmVudENvbG9yIj48L3BhdGg%2BPC9zdmc%2B&widths=undefined&widths=auto&heights=250&heights=150"
```
Comment (leerob) [inline app/head.tsx:12]: Do you want to use https://vercel.com/blog/introducing-vercel-og-image-generation-fast-dynamic-social-card-images instead? We are not actively using this service anymore.

**Review by leerob — COMMENTED** [review]
Code is looking really good!

File: app/head.tsx (hunk 1 above)
Comment (stipsan) [reply app/head.tsx:12]: Sure, been waiting for an excuse to try it out :D

Comment (georgeosutton) [reply app/head.tsx:16]: Would be good to combo this suggestion with a preview pane in Studio.

Comment (stipsan) [reply app/head.tsx]: Added 👍 
image

**Pull request merged**